### PR TITLE
feat: prepare ncu for new README format

### PR DESCRIPTION
**Pull request by Trott**
This accommodates both the current README format and the upcoming README
format.

Refs: https://github.com/nodejs/node/pull/40137

### CODE REVIEW

File: lib/collaborators.js [hunk 1]
```diff
@@ -6,7 +6,8 @@ const TSC_TITLE = '### TSC (Technical Steering Committee)';
 const TSCE_TITLE = '### TSC Emeriti';
 const CL_TITLE = '### Collaborators';
 const CLE_TITLE = '### Collaborator Emeriti';
-const CONTACT_RE = /\* +\[(.+?)\]\(.+?\) +-\s\*\*(.+?)\*\* +&lt;(.+?)&gt;/mg;
+const CONTACT_RE =
+  /\* +\[(.+?)\]\(.+?\) +-\s\*\*(.+?)\*\* +(?:&lt;|<)(.+?)(?:&gt;|>)/mg;
```
Comment (targos) [inline lib/collaborators.js:10]: don't we need to include the backslash in the regexp?

Comment (targos) [reply lib/collaborators.js:10]: I mean, we clearly don't, because the tests pass, but I don't understand how it works ^^

Comment (Trott) [reply lib/collaborators.js:10]: I'm not sure I understand which backslash you are talking about?

Comment (targos) [reply lib/collaborators.js:10]: The one you had to add in the readme to escape `<`

Comment (Trott) [reply lib/collaborators.js:10]: It doesn't work. I thought I ran it against the main branch, but I guess I didn't.

Comment (Trott) [reply lib/collaborators.js]: Nice catch, @targos. Fixed.

**Review by targos — APPROVED** [review]

**Comment by Trott** [comment]
Do we know why the 16.x GitHub Action is failing on a seemingly-unrelated test?

**Comment by targos** [comment]
The tests are a bit flaky on Windows

**Comment by Trott** [comment]
> The tests are a bit flaky on Windows

OK, I'll re-run it. Thanks.

**Pull request merged**